internal/datacoord/meta.go

```diff
@@ -780,15 +780,15 @@ func (m *meta) SetSegmentCompacting(segmentID UniqueID, compacting bool) {
 }
 
 func (m *meta) CompleteMergeCompaction(compactionLogs []*datapb.CompactionSegmentBinlogs, result *datapb.CompactionResult,
-	canCompaction func(segment *datapb.CompactionSegmentBinlogs) bool) error {
+	canCompaction func(segment *datapb.CompactionSegmentBinlogs) bool) (bool, error) {
```

Comment:
return bool and error at same time is confusing. maybe at L791 return a special error and let the caller handle the error is better

Reply:
good idea, I will do it.

Reply:
I have fixed it, please review @jiaoew1991